controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -73,6 +73,106 @@ static bool validate_hex_notation(const std::string &str, uint8_t expected_octet
     return true;
 }
 
+/**
+ * @brief Parse bss_info string into bss_info_conf_struct.
+ * 
+ * @param[in] bss_info_str String conataining bss info configuration
+ * @param[out] bss_info_conf Controller database struct filled with configuration.
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+static std::string parse_bss_info(const std::string &bss_info_str,
+                                  db::bss_info_conf_t &bss_info_conf, std::string &err_string)
+{
+    auto confs = string_utils::str_split(bss_info_str, ' ');
+    if (confs.size() != 8) {
+        err_string = "missing configuration";
+        return std::string();
+    }
+
+    // Alid
+    std::string al_mac(confs[0]);
+    if (!net::network_utils::is_valid_mac(al_mac)) {
+        err_string = "invalid al_mac";
+        return std::string();
+    }
+
+    bss_info_conf = {};
+
+    // Operating class
+    uint operating_class = string_utils::stoi(confs[1]);
```

Comment:
If you look at Table 11, you see that operating class can be "8x", "11x" or "12x". At the end of section 2.6.1, these strings are explained: 8x stands for 81 or 83 or 84, 11x stands for 115 or 116, and 12x stands for 124, 125 or 126. I haven't checked what these map on exactly, but I suppose 8x == 2.4GHz, 11x == 5GHz low, and 12x == 5GHz high.

To be fully flexible, the database should keep a list of operating classes in bss_info_conf. However, I think we can probably suffice with just putting 2.4 or 5GHz. Do we already have an enum for this?

Reply:
Adding an operating class table will take place as part of channel selection full flow #62.
It is not true how you have divided the operating classes. 
For example OpClass 128 has channels: {42,58,106,122,138,155}.
Since I'm not sure what exact bss_info will send us, I prefer to keep it that way to have flexibility. What you proposed will reduce flexibility.

Reply:
It is very clearly stated in 3 different places that this can be 8x, 11x or 12x. So string_utils::stoi is certainly wrong.

For reference: the three examples in section 10.10 of the CAPI spec have 8x; section 2.6.1 of the EasyMesh test plan explains the meaning of the three possible values; and section 3.3.1, Table 11 gives all possible bss_info strings, and it's always one of these three.